SDWebImage/SDAnimatedImageView.m

```diff
@@ -123,6 +124,11 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder
 
 - (void)commonInit
 {
+    if (self.hasInitialized) {
```

Comment:
Why does this place need to be marked using `hasInitialized`? 
@dreampiggy If you have time, you could add some explains with `initWithImage:`,  Thanks.

Reply:
Using another way to solve this. And also with comments there. Please re-review